### ISSUE TITLE
seeded data generators should return to original random state

### ISSUE THREAD

**Issue by Happyzippy** [keras-team/keras#8496]
Seeding of data generators introduce side effects because the random state is not reset after use

The following code should not make c and d deterministic, and not equal either
`
from keras.preprocessing.image import ImageDataGenerator
import numpy as np
x = np.random.randn(10,10,10,3)
g2 = ImageDataGenerator().flow(x, seed=0, shuffle=True)
g1 = ImageDataGenerator().flow(x, seed=0, shuffle=True)

a = next(g1)
c = np.random.randn(10)
b = next(g2)
d = np.random.randn(10)
assert np.array_equal(c,d) == False
`

**Comment by ozabluda** [comment]
For more (friendly) criticism of Keras' usage of numpy.random.seed() see https://github.com/numpy/numpy/pull/9906#discussion_r146420055

**Comment by nosound2** [comment]
Given the described above behavior, in case of multi-threading using the seeds with ImageDataGenerator is outright useless. The race condition to set and use the global numpy seed produces non-deterministic results.
